specification/compute/resource-manager/Microsoft.Compute/GalleryRP/stable/2022-08-03/galleryRPCommon.json

```diff
@@ -0,0 +1,222 @@
+          "type": "string",
+          "description": "The plan ID."
+        },
+        "publisher": {
+          "type": "string",
+          "description": "The publisher ID."
+        },
+        "product": {
+          "type": "string",
+          "description": "The product ID."
+        }
+      },
+      "description": "Describes the gallery image definition purchase plan. This is used by marketplace images."
+    },
+    "SharedGalleryImageVersionStorageProfile": {
+      "type": "object",
+      "properties": {
+        "osDiskImage": {
+          "$ref": "#/definitions/SharedGalleryOSDiskImage"
+        },
+        "dataDiskImages": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SharedGalleryDataDiskImage"
+          },
+          "x-ms-identifiers": [
+            "lun"
+          ],
+          "description": "A list of data disk images."
+        }
+      },
+      "description": "This is the storage profile of a Gallery Image Version."
+    },
+    "SharedGalleryOSDiskImage": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/SharedGalleryDiskImage"
+        }
+      ],
+      "description": "This is the OS disk image."
+    },
+    "SharedGalleryDataDiskImage": {
+      "type": "object",
+      "properties": {
+        "lun": {
+          "type": "integer",
+          "format": "int32",
+          "description": "This property specifies the logical unit number of the data disk. This value is used to identify data disks within the Virtual Machine and therefore must be unique for each data disk attached to the Virtual Machine."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/SharedGalleryDiskImage"
+        }
+      ],
+      "required": [
+        "lun"
+      ],
+      "description": "This is the data disk image."
+    },
+    "SharedGalleryDiskImage": {
+      "type": "object",
+      "properties": {
+        "diskSizeGB": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "description": "This property indicates the size of the VHD to be created."
+        },
+        "hostCaching": {
+          "type": "string",
+          "description": "The host caching of the disk. Valid values are 'None', 'ReadOnly', and 'ReadWrite'",
+          "enum": [
+            "None",
+            "ReadOnly",
+            "ReadWrite"
+          ],
+          "x-ms-enum": {
+            "name": "SharedGalleryHostCaching",
+            "modelAsString": true
+          }
+        }
+      },
+      "description": "This is the disk image base class."
+    },
+    "Architecture": {
+      "type": "string",
+      "description": "The architecture of the image. Applicable to OS disks only.",
+      "enum": [
+        "x64",
+        "Arm64"
+      ],
+      "x-ms-enum": {
+        "name": "Architecture",
+        "modelAsString": true
+      }
+    }
+  },
+  "parameters": {
+    "LocationNameParameter": {
```

Comment:
>LocationNameParameter [](http://example.com/codeflow?start=5&length=21)

You can reference to the location parameter from common-types types.json instead.

Reply:
@TimLovellSmith , while that may be true, that is outside the scope of this release as this is in the previously released versions.